sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2554,6 +2617,10 @@ pi_int32 ExecCGCommand::enqueueImp() {
       std::sort(std::begin(ReqToMem), std::end(ReqToMem));
     }
 
+    // Host task is executed asynchronously so we should record where it was
+    // submitted to report exception origin properly.
+    copySubmissionCodeLocation();
```

Comment:
Can this not be done in the base class? The base class constructor is also getting "CodeLoc" from the submit call, so you may be able to consolidate this and call copySubmissionCodeLocation() there.

Reply:
I did it here because now it is the only type of command that deal with it. And no need to do this operation for all other classes.